src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java

```diff
@@ -185,42 +200,57 @@ public void stop(Federation federation) {
 
     @PreDestroy
     public void tearDown() {
-        org.bitcoinj.core.Context.propagate(new org.bitcoinj.core.Context(ThinConverter.toOriginalInstance(bridgeConstants.getBtcParamsString())));
+        org.bitcoinj.core.Context.propagate(
+            new org.bitcoinj.core.Context(
+                ThinConverter.toOriginalInstance(bridgeConstants.getBtcParamsString())));
         peerGroup.stop();
         peerGroup = null;
     }
 
     private class BtcReleaseEthereumListener extends EthereumListenerAdapter {
         @Override
         public void onBestBlock(org.ethereum.core.Block block, List<TransactionReceipt> receipts) {
+            if (!isPegoutEnabled) {
+                return;
+            }
+
             boolean hasBetterBlockToSync = nodeBlockProcessor.hasBetterBlockToSync();
             boolean isStorageSynced = storageSynchronizer.isSynced();
             if (hasBetterBlockToSync || !isStorageSynced) {
                 logger.trace(
-                    "[onBestBlock] Node is not ready to process pegouts. hasBetterBlockToSync: {} isStorageSynced: {}",
+                    "[onBestBlock] Node is not ready to process pegouts. hasBetterBlockToSync: {} - isStorageSynced: {}",
                     hasBetterBlockToSync,
                     isStorageSynced
                 );
                 return;
             }
+            storageSynchronizer.processBlock(block, receipts);
```

Comment:
you sure it is okay to move this to before getting the federator state?

Reply:
It should be, I actually think we don't use it. Maybe @marcos-iov can correct me if I am wrong.

Reply:
Getting the federator state is not related to processing blocks and receipts by the storage synchronizer.  But it must be before `processReleases`.